exit pyempaq using the unpacked script exit code

ainquel opened the pull request:
Hello @facundobatista,

I'm using `pyempaq` to pack a cli and wanted to be able to check its meaningful exit code so I can create shell scripts around it.
Currently when my cli crashes, I can see the traceback (as stdout/stderr are not captured) but it always exits with code `0`.

I would like to know if you are agree to return the exit code of the unpacked script ?

This is what this PR is currently doing.

I thought also that maybe you would want to differentiate unpack errors from user's script ones using a special code ? (there is one `exit(1)` when `unpack_restrictions` are not ok)

Seems also related to #64 

(thank you for this useful project !)

facundobatista commented:
Hello @ainquel !

Thanks for this PR, I agree on most of the things you say. And it's quite solid!

Yes, we need to differentiate unpack errors from user's script ones. What about creating an "internal rule" like "unpacker itself when exiting on errors will return codes above 64 to minimize the possibility of clashing with return codes from the payload"?.

Beyond that, things missing in this PR (that you can improve it here or I would do later):

- a specific unit test for `run_command` (in `tests/test_unpacker.py`), asserting that it returns the subprocess' returncode.
- a note in the README informing that PyEmpaq transmits the payload's returncode (the feature you're adding, I mean)
- regarding above consideration, change the current `exit(1)` to `exit(64)` and add a note in the README specifying which returncodes the *unpacker itself* uses (and for what)

Let me know how you want to proceed! 

Thanks again!

ainquel commented:
Yes the internal rule looks like a good idea to me.

I've pushed changes based on your suggestions and some additionals.

- I've changed the signature of `run_command` to return a `CompletedProcess` instead of the exit code, it feels better to receive a process when calling this method. If later, there is a need of inspecting stdout or stderr, the signature will remain the same
- I've refactored some code in `test_integration.py` that was common to all of the tests

Let me know if it seems correct to you !

facundobatista reviewed (approved):
Wonderful, thanks for all this work!!
